workspace/workflows/jobs.py

```diff
@@ -95,21 +113,52 @@ def remove_workflows_skipped_on_main(self, workflows):
         for workflow_id in skipped:
             workflows.pop(workflow_id, None)
 
-    def get_all_runs(self) -> list:
+    def get_runs_since_last_retrieval(self) -> list:
         params = {"branch": self.branch} if self.branch else {}
         params["per_page"] = 100
+        if self.last_retrieval_timestamp:  # If not present do not pass anything at all
+            params["created"] = ">=" + self.last_retrieval_timestamp
         return self._get_json_response("actions/runs", params=params)["workflow_runs"]
 
     def get_latest_conclusions(self) -> dict:
-        all_runs = self.get_all_runs()
-        latest_runs, missing_ids = self.find_latest_for_each_workflow(all_runs)
+        """
+        Use the GitHub API to get the conclusion of the most recent run for each workflow.
+        Update the cache with the conclusions and the timestamp of the retrieval.
+        """
+        # Use the moment just before calling the GitHub API as the timestamp
+        timestamp = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        new_runs = self.get_runs_since_last_retrieval()
+        latest_runs, missing_ids = self.find_latest_for_each_workflow(new_runs)
         conclusions = {
             run["workflow_id"]: self.get_conclusion_for_run(run) for run in latest_runs
         }
-        missing = {workflow_id: "missing" for workflow_id in missing_ids}
-        conclusions.update(missing)
+        self.fill_in_conclusions_for_missing_ids(conclusions, missing_ids)
+
+        self.cache = {
+            "timestamp": timestamp,
+            # To be consistent with the JSON file which has the IDs as strings
+            "conclusions": {str(k): v for k, v in conclusions.items()},
+        }
         return conclusions
 
+    def fill_in_conclusions_for_missing_ids(self, conclusions, missing_ids):
+        """
+        For workflows that have not run since the last retrieval, use the conclusion from the cache.
+        If no conclusion is found in the cache, mark the workflow as missing.
+        """
+        previous_conclusions = self.cache.get("conclusions", {})
+        for workflow_id in missing_ids:
+            id_str = str(workflow_id)  # In the cache JSON, IDs are stored as strings
+            conclusions[workflow_id] = previous_conclusions.get(id_str, "missing")
+        return
+
+    def update_cache_file(self):
+        master_cache = load_cache()
```

Comment:
Can we name this `main_cache` or `all_repos_cache` to avoid the word "master"?

Reply:
I have called it `cache_file_contents` to be _kind of_ consistent with the tests.